chapters/statements.tex

```diff
@@ -430,6 +431,25 @@ \subsubsection{Restrictions on When-Statements}\label{restrictions-on-when-state
 \end{lstlisting}
 \end{example}
 
+\paragraph*{Restrictions on statements within when-statements:}\label{restrictions-on-statements-within-when-statements}
```

Comment:
Can you really label a starred title?  Aren't you going to get a label attached to something other than you expect?

I suppose this is where all the new `\paragraph*` come from.  If we want these to be labeled, perhaps a `\…section` command could work better?  As I think of these sectioning commands, it's not entirely clear what belongs to a titled paragraph; just paragraph containing the title, or everything until the next sectioning command?

```suggestion
\subsubsection{Restrictions on statements within when-statements:}\label{restrictions-on-statements-within-when-statements}
```